pkg/secrethub/me.go

```diff
@@ -8,15 +8,20 @@ type MeService interface {
 	Repos() ([]*api.Repo, error)
 	// User retrieves the current users details.
 	User() (*api.User, error)
+	// SendVerificationEmail sends an email to the authenticated user's registered email address
+	// for them to prove they own that email address.
+	SendVerificationEmail() error
 }
 
 type meService struct {
+	client      client
 	repoService RepoService
 	userService UserService
 }
 
-func newMeService(repoService RepoService, userService UserService) MeService {
+func newMeService(client client, repoService RepoService, userService UserService) MeService {
```

Comment:
why add the `repoService` and `userService` as arguments here? What's keeping you from constructing those services inside the `newMeService` function?

Reply:
Because by doing so, the me service only depends on the interfaces, not on the private struct. For now the repo- and userservice can be constructed with only the client as argument, but what if that changes?

One could also argue the other way around. By adding the repo- and userservice as args of newMeService, we expose that the meService uses these internally.

I'm fine with both options.

Reply:
Which do you prefer?

Reply:
The struct field is the interface, the only surface area where the concrete implrementations touch would be inside the `newMeService` function. Seeing as the alternative is that the exact same surface area would exist in the `client.Me()` function, that's fine. 

When you want to create a custom `MeService` with internal interfaces, you can just create the struct yourself directly.